gentoopm/basepm/pkg.py

```diff
@@ -417,6 +420,13 @@ def maintainers(self):
         """
         pass
 
+    @property
+    @abstractmethod
+    def upstream(self) -> typing.Optional[PMUpstream]:
+        """
+        Get the package upstream metadata (or ``None`` if unavailable).
+        """
```

Comment:
I'm wondering if the distinction for `None` is really useful here. After all, none of the "upstream" keys are mandatory, so there's no technical difference between "missing" upstream metadata and "empty" upstream metadata, so it feels like asking consumers to separately detect two equivalent cases.

Reply:
I reduced the number of keys with such distinction